rfcs/XXXX-gene-sets/text.md

```diff
@@ -0,0 +1,97 @@
+
+**Approvers:** [Friendly Tech Person](mailto:arathi.mani@chanzuckerberg.com), [Friendly PM Person](mailto:schambers@chanzuckerberg.com), [Friendly Tech Person2](mailto:colin.megill@chanzuckerberg.com), [Girish Patangay](mailto:girish.patangay@chanzuckerberg.com), [Sara Gerber](mailto:sara.gerber@chanzuckerberg.com)
+
+## tl;dr
+
+This RFC contains the details about the data format for the gene set feature. It will support the import and export of differential
+expression results and allow a user to upload their own gene sets for analysis.
+
+## Glossary
+
+- Gene set - List of genes
+- Differential expression - A statistic (typically logfold and p-value) describing the difference in gene expression between two sets of cells. It may refer to stats specific to one gene (differential expression value) or many (differential expression analysis). When referring to many genes it is typically capped (eg top 10 most differentially expressed genes)
+- LogFold - A ratio of the difference in expression levels. Typically calculated by taking the log2 of the gene expression value for each cell and then finding the mean of that value for each set of cells. Because the data is already log2 transformed you can then take the difference between the two means as the logfold value
+
+## Problem Statement | Background
+
+Gene sets are the expected output of differential expression analysis in cellxgene. Two (non-overlaping) sets of cells
+are selected and the difference in the expression level for each gene is computed by comparing the average
+expression level for each set of cells. When there is a particularly large difference in expression between the two sets of
+cells, that gene is identified and presented to the user. To ensure consistency and compatibility it is necessary to define a
+data format for storing references to these genes and any accompanying statistical data.
+
+Product Requirements
+
+### Users should be able to export differential expression results
+
+Currently cellxgene users can choose two sets of cells and compute/display the log fold and adjusted p-value of the top ten differentially expressed genes. Additionally we display a histogram of the gene expression levels of the selected cells for each gene in the right side bar.
+
+- [US1] A user should be able to export a gene set, if the gene set was created by differential expression, the pvalue, log fold change, and category label fields will be populated in addition to the gene name. This will be available in a csv file (see data formats below for more detail).
+
+### Users should be able to upload their own gene sets and see them in the right side bar with the option to apply them within current cellxgene functionalities, if the same gene ontology was used when generating the current dataset and the geneset uploaded by the user (ie the gene names match)
+
+- [US2] A user should be able to upload a geneset via a csv file (see data formats below for more detail) for a dataset.
```

Comment:
Are any of these upload related requirements addressed in this RFC?

Reply:
This RFC is strictly focused on the data format. 
RE upload -- We should be able to keep things simple and limit files to a size that can be uploaded in the browser using a common library.

Reply:
I assume all of the requirements will be met by the end of reading the RFC, but I don't see anything about how the upload is achieved. I would avoid adding requirements that are not specifically addressed in your RFC. It keeps the doc focused on the problem you're actually solving.